packages/api/src/webhooks/cannon.ts

```diff
@@ -402,36 +407,84 @@ export default class WebhookCannon {
         );
         // we don't retry on non 400 responses. only on timeouts
         // this.retry(event);
-        return;
       } catch (e) {
         console.log("firing error", e);
+        errorMessage = e.message;
         await this.retry(trigger, params, e);
+      } finally {
+        await this.storeResponse(webhook, event, resp, startTime, responseBody);
+        await this.storeTriggerStatus(
+          trigger.webhook,
+          triggerTime,
+          statusCode,
+          errorMessage
+        );
         return;
       }
     }
   }
 
+  async storeTriggerStatus(
+    webhook: DBWebhook,
+    triggerTime: number,
+    statusCode?: number,
+    errorMessage?: string
+  ) {
+    try {
+      if (statusCode >= 300 || !statusCode) {
+        await this.db.webhook.updateStatus(webhook.id, {
+          status: {
+            lastTriggeredAt: triggerTime,
+            lastFailure: {
+              timestamp: triggerTime,
+              statusCode: statusCode,
+              errorMessage: errorMessage,
```

Comment:
I think we're missing the separate `responseBody` vs `errorMessage` here. Do you think it will be simpler with only a single field? I'm ok with that as well.

Reply:
I think it's better to have separate fields as well, now there're error and response instead of only errorMessage!